plugin-server/tests/worker/ingestion/lazy-person-container.test.ts

```diff
@@ -0,0 +1,59 @@
+import { DateTime } from 'luxon'
+
+import { Hub } from '../../../src/types'
+import { createHub } from '../../../src/utils/db/hub'
+import { UUIDT } from '../../../src/utils/utils'
+import { LazyPersonContainer } from '../../../src/worker/ingestion/lazy-person-container'
+import { resetTestDatabase } from '../../helpers/sql'
+
+jest.mock('../../../src/utils/status')
+
+const timestamp = DateTime.fromISO('2020-01-01T12:00:05.200Z').toUTC()
+const uuid = new UUIDT()
+
+describe('LazyPersonContainer()', () => {
+    let hub: Hub
+    let closeHub: () => Promise<void>
+    let personContainer: LazyPersonContainer
+
+    beforeEach(async () => {
+        ;[hub, closeHub] = await createHub()
+        await resetTestDatabase()
+        personContainer = new LazyPersonContainer(2, 'my-id', hub)
+
+        jest.spyOn(hub.db, 'fetchPerson')
+    })
+
+    afterEach(async () => {
+        await closeHub()
+    })
+
+    it('.get returns undefined if person does not exist', async () => {
+        const persons = await Promise.all([personContainer.get(), personContainer.get(), personContainer.get()])
+
+        expect(persons).toEqual([undefined, undefined, undefined])
+        expect(personContainer.loaded).toEqual(false)
+        expect(hub.db.fetchPerson).toHaveBeenCalledTimes(1)
```

Comment:
Not sure what you mean by this comment. A container only lives for the lifecycle of a single event so cross-thread behavior doesn't matter here.

Reply:
This is probably fine based on other assumptions we make in the code/error handling.

Imagine we have two plugin-servers executing in parallel: on distinct_id1 and distinct_id2 respectively neither of which exist before. They both create the lazypersoncontainers and run the first get - both now keep the state that the person doesn't exist. Next assume thread 1 handles an alias call, which creates a person with these both ids. Thread 2 will still `.get()` undefined instead of fetching again.

Reply:
Nothing new introduced with LazyPersonContainers here - the exact same behavior happens when you fetch consistently or what we did previously. 

The fix is ordering, buffering and deprecating $create_alias if possible.